tensorboard/plugins/base_plugin.py

```diff
@@ -245,12 +246,12 @@ def __init__(
           function is cheap. The returned connection must only be used by a
           single thread. Things like connection pooling are considered
           implementation details of the provider.
-      db_uri: The string db URI TensorBoard was started with. If this is set,
-          the logdir should be None.
+      db_uri: The string db URI TensorBoard was started with.
       flags: An object of the runtime flags provided to TensorBoard to their
           values.
-      logdir: The string logging directory TensorBoard was started with. If this
-          is set, the db_uri should be None.
+      logdir: The string logging directory TensorBoard was started with.
+      logdir_spec: The value of `--logdir_spec` passed to TensorBoard (see
```

Comment:
Let's just have the data location handler consult `context.flags.logdir_spec` directly, even though it's uglier, to avoid presenting logdir_spec as something that plugin implementers can depend on.  (If we did want them to use it we'd probably want to pass it to them in parsed form, not the colon and comma delimited form.)